### PR TITLE
Don't show invalid ABI interfaces

### PR DESCRIPTION
**Pull request by ngotchac**
And don't show execute button on contract view if no functions to execute from the contract

### CODE REVIEW

File: js/src/util/abi.js [hunk 1]
```diff
@@ -147,3 +149,31 @@ export function parseAbiType (type) {
...
+    return new Abi(_abi);
+  } catch (error) {
+    // If there is a wrong type, filter it out
+    if (/Cannot convert .+ valid ParamType/i.test(error.message)) {
+      const validInterfaces = [];
+
+      _abi.forEach((interf) => {
+        try {
+          Abi.parseABI([ interf ]);
+          validInterfaces.push(interf);
```
Comment (jacogr) [inline js/src/util/abi.js]: A filter would probably have been more appropriate since we are filtering?

File: js/src/api/contract/contract.js [hunk 2]
```diff
@@ -29,7 +29,9 @@ export default class Contract {
     }
 
     this._api = api;
-    this._abi = new Abi(abi);
+    this._abi = abi instanceof Abi
```
Comment (jacogr) [inline js/src/api/contract/contract.js]: I prefer the original approach where we are explicit on what we expect in as opposed to having either one or the other? No other way around it?

File: js/src/util/abi.js [hunk 3]
```diff
@@ -147,3 +149,31 @@ export function parseAbiType (type) {
...
+  try {
+    return new Abi(_abi);
+  } catch (error) {
+    // If there is a wrong type, filter it out
+    if (/Cannot convert .+ valid ParamType/i.test(error.message)) {
+      const validInterfaces = [];
+
+      _abi.forEach((interf) => {
+        try {
+          Abi.parseABI([ interf ]);
```
Comment (jacogr) [inline js/src/util/abi.js]: When doe this become a problem and why is this ont on the actual ABI layer instead of just benefitting the UI?

**Comment by jacogr** [comment]
Tests failing?

**Comment by debris** [comment]
@paritytech/ui-devs any updates on this?

**Pull request closed without merging**